packages/trader/src/App/Containers/trade-settings-extensions.jsx

```diff
@@ -43,23 +43,14 @@ class TradeSettingsExtensions extends React.Component {
         populateSettingsExtensions(menu_items);
     };
 
-    componentDidMount() {
-        this.populateSettings();
-    }
+    React.useEffect(() => {
+        populateSettings();
 
-    componentDidUpdate() {
-        this.populateSettings();
-    }
+        return () => populateSettingsExtensions(null);
+    });
```

Comment:
can we add dependency array here?

Reply:
We pass the whole store to `MobxContentProvider` in `populateSettings` function, so I think it’s better if we don’t add dependencies here.

Reply:
but previously `populateSettingsExtensions` was called on `componentWillUnmount`. Now, it is called on each cleanup. I think we should have a separate `useEffect` for this.